DQM/src/DQM/DarkBremInteraction.cxx

```diff
@@ -83,31 +83,29 @@ void DarkBremInteraction::produce(framework::Event& event) {
      *    exhausted the maximum number of tries to get a dark brem
      *    to occur. We just leave early so that the entries in the
      *    ntuple are the unphysical numeric minimum.
-     *
-     * This can also happen during development, so I leave a debug
-     * printout here to be uncommented when developing the dark
-     * brem simulation.
-    std::cout << "Event " << e.getEventNumber()
-      << " did not have a dark brem occur within it." << std::endl;
      */
+    ldmx_log(error) << " No dark brem occured in this event";
     return;
   }
 
   if (recoil == nullptr or aprime == nullptr or beam == nullptr) {
     // we are going to end processing so let's take our time to
     // construct a nice error message
-    std::stringstream err_msg;
-    err_msg
+    ldmx_log(fatal)
         << "Unable to find all necessary particles for DarkBrem interaction."
-        << " Missing: [ " << (recoil == nullptr ? "recoil " : "")
-        << (aprime == nullptr ? "aprime " : "")
-        << (beam == nullptr ? "beam " : "") << "]" << std::endl;
-    EXCEPTION_RAISE("BadEvent", err_msg.str());
+        << " Missing: [ " << (recoil == nullptr ? " recoil " : "")
+        << (aprime == nullptr ? " aprime " : "")
+        << (beam == nullptr ? " beam " : "") << "]";
+    EXCEPTION_RAISE(
+        "BadEvent",
+        "Unable to find all necessary particles for DarkBrem interaction.");
     return;
   }
 
   const auto& recoil_p = recoil->getMomentum();
   const auto& aprime_p = aprime->getMomentum();
+  TVector3 recoil_pvec(recoil_p[0], recoil_p[1], recoil_p[2]);
+  TVector3 aprime_pvec(aprime_p[0], aprime_p[1], aprime_p[2]);
```

Comment:
ROOT's docs encourage us to use `ROOT::Math::XYZVector` instead

> [TVector3](https://root.cern.ch/doc/master/classTVector3.html) is a legacy class. It is slower and worse for serialization than the recommended superior alternative [ROOT::Math::XYZVector](https://root.cern.ch/doc/master/namespaceROOT_1_1Math.html#a91b80e54b44a65c90d60e5c8ff128746).

https://root.cern.ch/doc/master/classTVector3.html

They have a whole Physics Vector page to help folks with the updated interface: https://root.cern.ch/doc/master/group__GenVector.html